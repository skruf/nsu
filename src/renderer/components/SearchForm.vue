<style lang="stylus">
.search-form
  display flex
  margin 10px 0 20px
  .el-button
    margin-left 20px
</style>

<template>
  <el-form ref="form" class="search-form" @submit.prevent.native="submit">
    <el-input
      v-model="v"
      @input="input"
      @clear="submit"
      :placeholder="placeholder"
      :clearable="true"
    ></el-input>

    <el-button native-type="submit" type="primary">Search</el-button>
  </el-form>
</template>

<script>
export default {
  name: "SearchForm",
  props: {
    value: { type: String, default: "" },
    placeholder: { type: String, default: "Enter search words.." }
  },
  data: function() {
    return {
      v: this.value
    }
  },
  methods: {
    input() {
      this.$emit("input", this.v)
      if(this.v === "") {
        this.submit()
      }
    },
    submit() {
      this.$emit("submit", this.v)
    }
  }
}
</script>
