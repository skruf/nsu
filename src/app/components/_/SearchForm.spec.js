import createLocalVue from "~/utils/tests/createLocalVue"
import mount from "~/utils/tests/mount"
import Element, { Form, Input } from "element-ui"
import SearchForm from "~/components/SearchForm"

const localVue = createLocalVue([ Element ])

const build = (deep = false, overrides = {}) => {
  return mount(SearchForm, localVue, overrides, deep)
}

describe("SearchForm", () => {
  it("should render the component", () => {
    const wrapper = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have a form and an input", () => {
    const wrapper = build()
    expect(wrapper.find(Form).exists()).toBe(true)
    expect(wrapper.find(Input).exists()).toBe(true)
  })

  it("has correct placeholder", () => {
    const expectedValue = "This is a test placeholder"
    const wrapper = build(false, {
      propsData: { placeholder: expectedValue }
    })
    const input = wrapper.find(Input)
    const currentValue = input.attributes("placeholder")
    expect(currentValue).toBe(expectedValue)
  })

  it("sets value when typing", () => {
    const expectedValue = "this is a test"
    const wrapper = build()
    const input = wrapper.find(Input)
    input.vm.$emit("input", expectedValue)
    expect(wrapper.emitted("input")[0][0]).toBe(expectedValue)
    expect(wrapper.vm.v).toBe(expectedValue)
  })

  it("submits form with value", () => {
    const expectedValue = "this is a test"
    const wrapper = build()
    wrapper.vm.v = expectedValue
    wrapper.vm.submit()
    const emittedValue = wrapper.emitted("submit")[0][0]
    expect(emittedValue).toBe(expectedValue)
  })

  it("submits form when value is empty", () => {
    const expectedValue = ""
    const wrapper = build()
    const input = wrapper.find(Input)
    input.vm.$emit("input", expectedValue)
    const emittedValue = wrapper.emitted("submit")[0][0]
    expect(emittedValue).toBe(expectedValue)
  })
})
