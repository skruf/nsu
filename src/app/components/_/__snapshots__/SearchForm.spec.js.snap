// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render the component 1`] = `
<el-form-stub labelsuffix="" showmessage="true" validateonrulechange="true" class="search-form">
  <el-input-stub value="" type="text" autocomplete="off" validateevent="true" prefixicon="el-icon-search" clearable="true" placeholder="Enter search words.."></el-input-stub>
</el-form-stub>
`;
